__tests__/ExpensiMark-HTML-test.js

```diff
@@ -102,10 +102,10 @@ test('Test markdown replacement for emails wrapped in bold/strikethrough/italic
     expect(parser.replace(testInput)).toBe('<strong><a href="mailto:abc@gmail.com">abc@gmail.com</a></strong>');
 
     testInput = '_abc@gmail.com_';
-    expect(parser.replace(testInput)).toBe('<em><a href="mailto:abc@gmail.com">abc@gmail.com</a></em>');
+    expect(parser.replace(testInput)).toBe('<a href="mailto:_abc@gmail.com">_abc@gmail.com</a>_');
```

Comment:
I don't think this would be expected behaviour and would result in a regression. If you wrap a normal email (`abc@gmail.com`) in italic markdown it should just be italicised and linked. cc @aimane-chnaif

Reply:
> @Antasel Let's add all of these in test case. Github markdown seems to have different logic
> 
> * `abc@gmail.com` -> [abc@gmail.com](mailto:abc@gmail.com)
> * `_abc@gmail.com` -> [_abc@gmail.com](mailto:_abc@gmail.com)
> * `__abc@gmail.com` -> [__abc@gmail.com](mailto:__abc@gmail.com)
> * `abc@gmail.com_` -> abc@gmail.com_
> * `abc@gmail.com__` -> abc@gmail.com__
> * `_abc@gmail.com_` -> _[abc@gmail.com](mailto:abc@gmail.com)_
> * `__abc@gmail.com_` -> __[abc@gmail.com](mailto:abc@gmail.com)_
> * `_abc@gmail.com__` -> _[abc@gmail.com](mailto:abc@gmail.com)__
> * `__abc@gmail.com__` -> **[abc@gmail.com](mailto:abc@gmail.com)**

those cases are already covered in https://github.com/Expensify/App/issues/17387#issuecomment-1630415664
so there is no need to restore, just leave it as it is. 
How you think ? @jjcoffee

Reply:
@aimane-chnaif you meant "the case of _ prefixed email inside italic markdown can be skipped"  is following case ? 

`__abc@gmail.com_` -> __abc@gmail.com_
after parsed, underscore is not contained in email
if so, it can be done by only moving autoemail rule after italic rule. no need extra tweak in italic rule
cc: @aimane-chnaif

Reply:
@Antasel Sorry I'm not clear, what would Do you mean the result for `__abc@gmail.com_` will be